source/Core/Threads/GUIThread.cpp

```diff
@@ -860,8 +860,8 @@ void startGUITask(void const *argument) {
   getTipRawTemp(1); // reset filter
   OLED::setRotation(getSettingValue(SettingsOptions::OrientationMode) & 1);
   uint32_t ticks = xTaskGetTickCount();
-  ticks += (TICKS_SECOND * 4); // 4 seconds from now
-  while (xTaskGetTickCount() < ticks) {
+  ticks += (TICKS_SECOND * getSettingValue(SettingsOptions::LOGOTime));
+  if (getSettingValue(SettingsOptions::LOGOTime) > 0 && xTaskGetTickCount() < ticks) {
```

Comment:
@Ralim
Thank you for the guidance.
Still, I am not sure about this part.

I would really like to have the function to distinguish between **animations** and **static pictures** in that the duration:
1. for the animations would last to the last frame ***+ 2 x 'gap-time'*** (without the option to set the duration) and
2. for static there would be the option to set the duration

I think I would know how to achieve the option to be dependent on static pictures if there was a *'function'* like `defined(MODEL_TS100)`&nbsp;=>&nbsp;***`defined(static picture)`***. 😅